### ISSUE TITLE
Problem with Optiga Trust X and nRF52840

### ISSUE THREAD

**Issue by stevenraemaekers** [Infineon/fwupd-secboot-optiga-trust#1]
Hi,

I'm trying to get the Optiga Trust X to perform some work when sending a command over BLE. The problem is that the Optiga Trust X seems to hang at "TransceiveAPDU", in specific, it waits for optiga_comms_status not to be OPTIGA_COMMS_BUSY any more (see CommandLib.c:469).

I found out it has something to do with RTC timers not being set up properly. 
When I run the work directly (without waiting for BLE commands) when starting up the chip, it runs fine. Over BLE, however, it fails, while it is the exact same function call. 

There is a function optiga_comms_event_handler that is called to change the status of the optiga chip to be non-busy. However, for some reason, this function never gets called when requesting to run the function with BLE. This function seems to be called from a timer, that's why I thought RTC might have something to do with it. It looks like BLE somehow hijacks the RTC timer callbacks and prevents the Optiga Trust X from changing its status?

Do you have any idea what could be wrong?

**Comment by ayushev** [comment]
Hi there, just to understand. You did try to run examples, and everything was fine, but when you run the code with BLE in parallel you observe problems, correct?

**Comment by stevenraemaekers** [comment]
Well, your examples use a bootloader, and you make some changes in the file nrf_bootloader_dfu_timers.c. In my case, I'm not using a bootloader, and I also do not use this file but instead I simply have a single app that directly calls the Optiga Trust X.

My suspicion is that the changes you make in nrf_bootloader_dfu_timers.c with regards to the RTC are set up wrongly in my application, but I would not know where to make the proper changes.

**Comment by ayushev** [comment]
Sorry for stupid question, I meant whether you tried [examples](https://infocenter.nordicsemi.com/index.jsp?topic=%2Fcom.nordic.infocenter.sdk5.v15.3.0%2Flib_ifx_optiga.html) and then added some functionality like actual BLE communication and the code stoped working.
You think in the absolutely right direction (the original post), I just want to udnerstand the setup better.

**Comment by stevenraemaekers** [comment]
I started with a template example from Nordic and added Optiga
communication to that.
I took the code from your secure boot example repository.
The Nordic BLE communication was already fully functional before I added
Optiga code.

**Comment by ayushev** [comment]
I meant rather the examples available from SDK, not as part of this Application Note, but I guess you have tried them as well.

In you code, did you call 
```c
    // OPTIGA stack needs LF clock for RTC2
    if (!nrf_clock_lf_is_running())
    {
        nrf_clock_task_trigger(NRF_CLOCK_TASK_LFCLKSTART);
    }
```

This should enable RTC2 which is used by the Trust X host library stack.

**Comment by stevenraemaekers** [comment]
Yes, that has been called. In fact, communication with the Optiga Trust X
is successful as long as I do not communicate over BLE.
